src/content/posts/2023-11-17-auto-importing-components-in-astro-mdx.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Automatically import Components in Astro MDX
+slug: automatically-import-components-in-astro-mdx
+date: 2023-11-17 12:00:00
+tags:
+- frontend
+- Astro
+- SSG
+categories:
+- frontend
+twitter_text: How to automatically import components in Astro MDX files.
+authors: 
+- Ryan Trimble
+image: https://images.unsplash.com/photo-1419242902214-272b3f66ee7a?q=80&w=3913&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D
+image_url: https://unsplash.com/photos/silhouette-photo-of-mountain-during-night-time-ln5drpv_ImI
+image_credit: Vincentiu Solomon
+---
+
+One of my (very few) complaints while working with Astro was that using components in MDX files tends to be a bit clunky, especially for those who may not be as familiar with how to import things. For example, we have some non-developers that make content updates to our static sites who are most likely unaware of this crucial step.
+
+## The Problem
+
+We are converting sites from Hugo to Astro and in Hugo there is a concept of "shortcodes" used inside markdown to easily generate snippets of code. Hugo automatically knows about shortcodes, as long as they reside inside the `shortcodes/` folder, in order to be used throughout a markdown file without having to do any imports. 
+
+Astro doesn't have a "shortcodes" concept per se, but components are a pretty straightforward replacement. There are two main differences though:
+
+- Regular markdown does not understand components, which leads us to utlize MDX instead.
+- Components are not automatically imported into these MDX files.
+
+Now importing components may not sound like the end-of-the-world, but might cause a lot of unneeded stress the first time someone unknowingly uses a component without importing it first.
+
+## The Solution
+
+Over on Mastodon, Roma Komarov had the answer with a recent post called [Astro MDX Components](https://blog.kizu.dev/astro-mdx-components/). 
```

Comment:
```suggestion
Over on Mastodon, [Roma Komarov](https://front-end.social/@kizu) had the answer with a recent post called [Astro MDX Components](https://blog.kizu.dev/astro-mdx-components/). 
```
if you're going to call him out, you should link to him 😄

Reply:
good catch, added!